core/io/ip.cpp

```diff
@@ -129,6 +129,8 @@ IPAddress IP::resolve_hostname(const String &p_hostname, IP::Type p_type) {
 
 PackedStringArray IP::resolve_hostname_addresses(const String &p_hostname, Type p_type) {
 	List<IPAddress> res;
+
+#ifdef THREADS_ENABLED
```

Comment:
This change would skip the resolve cache in the no-threads case. Is that intended? Moreover, I'd say this file should compile and work well with the void implementations of mutex and friends.

Reply:
It was the first file that I worked on and never looked back. So yes, I'll try it with our modifications to mutex and such.

Reply:
Mmm. I'm currently skeptic. We don't currently run the "Thread.start" code, which loops through the IP adresses to resolve. So I'm afraid the code will get stuck.

Reply:
Nevermind. It works!